### ISSUE TITLE
Option to not set background color at all

### ISSUE THREAD

**Issue by hedyhli** [jdtsmith/kind-icon#38]
Thanks for this package, but like #30 I wish it was possible to have background color on the icons change together with the selection background.

### Transparent SVG

I understand that there may be limitations for kind-icons to know which candidate is currently selected in order to set its background differently, but is it possible not have background on those icons at all? I attempted to patch this in both kind-icon.el and also svg-lib, the best I could do made the icons white, somehow. The cached SVG files also have a white background, even though I'm sure the pictogrammers API supports fetching transparent SVGs.

Is this a limitation with emacs? Even if svg-lib could fetch the icons as transparent, I read somewhere that emacs would set its background to the frame background colors, so when the completion selection is active, its background would be stuck as the frame background.

### Text icons with no backgrond

Assuming fetching transparent SVGs is impossible, is it possible then for kind-icon to make text-icons not set a background color? I tried setting `kind-icon-default-face` to have `:background nil`, but the frame/text background colors is still applied. It should be possible to have text "icons" adapt to current selection background right? If so, how can I configure it?

As of now I have to edit kind-icon.el to achieve it:

```diff
diff --git a/kind-icon.el b/kind-icon.el
index 9ea16d8..d06c17f 100644
--- a/kind-icon.el
+++ b/kind-icon.el
@@ -319,7 +319,7 @@ background-color."
 				kind-face-bg
 			      default-bg)))
 		  (half (/ dfw 2))   ; integer division, may truncate
-		  (face-spec `(:weight bold :foreground ,col :background ,bg-col))
+		  (face-spec `(:weight bold :foreground ,col :background nil))
 		  (pad-right (propertize " " 'display `(space :width (,half))
 					 'face face-spec))
 		  (pad-left (propertize " " 'display `(space :width (,(- dfw half)))
@@ -335,7 +335,7 @@ background-color."
 			     ;; icon: always 2x1, half-space on each side
 			     (propertize ; pretend it's one char to allow padding
 			      (concat pad-left
-				      (propertize "*" 'display icon 'face `(:background ,bg-col))
+				      (propertize "*" 'display icon 'face `(:background nil))
 				      pad-right))
 			   ;; text, 1 or 2 chars, centered with full or half space on each side
 			   (let* ((txt (truncate-string-to-width (cadr map) 2))
```

<img width="500" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/50042066/a6081d3b-8202-487a-bed7-d7f6a414daec">

**Comment by jdtsmith** [comment]
This is good progress, thanks.  So it seems corfu will indeed apply a background to the kind-icon formatted text too, and it will display as long as it has none.  Emacs can certainly display transparent SVGs, but by default svg-lib insists on drawing a background rectangle behind the icon.  You can disable the rounded-rectangle "border" it draws with `:stroke 0` (as kind-icon does), but `:background nil` throws an error based on color conversion.  

What we need is for svg-lib to support drawing no background rectangle in `svg-lib-icon` if `background=nil`.  Then I can style the background only if `kind-icon-blend-background=t`, and otherwise leave it transparent.  I confirmed this works:

<img width="435" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/93749/5013e589-fbee-4712-a9c3-56ccf21d4919">

The fix for svg-lib is very easy; please open an issue there along these lines and report back.

```diff
diff -u /Users/jdsmith/.emacs.d/elpa/svg-lib-0.2.7/svg-lib.el /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-bMcNnT
--- /Users/jdsmith/.emacs.d/elpa/svg-lib-0.2.7/svg-lib.el	2023-07-10 07:50:19
+++ /var/folders/by/19mt78cj63v5_tslqh826g7w0000gn/T/buffer-content-bMcNnT	2023-09-12 10:24:56
@@ -212,10 +212,11 @@
   "Convert Emacs COLOR-NAME to #rrggbb form.
 If COLOR-NAME is unknown to Emacs, then return COLOR-NAME as-is."
   
-  (let ((rgb-color (color-name-to-rgb color-name)))
-    (if rgb-color
-        (apply #'color-rgb-to-hex (append rgb-color '(2)))
-      color-name)))
+  (when color-name
+    (let ((rgb-color (color-name-to-rgb color-name)))
+      (if rgb-color
+          (apply #'color-rgb-to-hex (append rgb-color '(2)))
+	color-name))))
 
 
 ;; SVG Library style build from partial specification
@@ -528,11 +529,12 @@
     (when (>= stroke 0.25)
       (svg-rectangle svg box-x box-y box-width box-height
                      :fill foreground :rx radius))
-    (svg-rectangle svg (+ box-x (/ stroke 2.0))
-                       (+ box-y (/ stroke 2.0))
-                       (- box-width stroke)
-                       (- box-height stroke)
-                       :fill background :rx (- radius (/ stroke 2.0)))
+    (when background
+      (svg-rectangle svg (+ box-x (/ stroke 2.0))
+                     (+ box-y (/ stroke 2.0))
+                     (- box-width stroke)
+                     (- box-height stroke)
+                     :fill background :rx (- radius (/ stroke 2.0))))
     
     (dolist (item (xml-get-children (car root) 'path))
       (let* ((attrs (xml-node-attributes item))
```

**Comment by jdtsmith** [comment]
Any luck on this?

**Comment by hedyhli** [comment]
kind-icon still draws a background for me after applying your patch for svg-lib even without the blend background setting. My guess is that it's either a problem with my config or changes are needed in kind-icon too; I highly suspect it's the former, but unfortunately I don't have time to test it further at the moment.

rougier recently replied on the svg-lib issue above requesting for PR. If you happen to have time, perhaps you could open the PR with your patch, considering that your setup worked (without modifying kind-icon)? Otherwise I will do the PR instead, after (hopefully) finding the fix for kind-icon still drawing the background.

**Comment by jdtsmith** [comment]
Great. I was waiting on this fix to release an implementation on the kind-icon side (which is simple).  You can try the `no-background` branch, which already supports background-less icons.  Please go ahead and turn the above diff to svg-lib.el into a PR, and check it with `no-background`.   I'm thinking of making this the default.

**Comment by hedyhli** [comment]
This branch works great, thanks!

**Comment by jdtsmith** [comment]
Closed in 4d26e8f.
